docs/execution/dataloader.rst

```diff
@@ -47,6 +46,20 @@ requested keys.
 A naive application may have issued *four* round-trips to a backend for the
 required information, but with ``DataLoader`` this application will make at most *two*.
 
+Note that loaded values are one-to-one with the keys and must have the same
+order. This means that if you load all values from a single query, you must
+make sure that you then order the query result for the results to match the keys:
+
+
+.. code:: python
+
+   class UserLoader(DataLoader):
+       def batch_load_fn(self, keys):
+           positions = {k: pos for pos, k in enumerate(keys)}
+           users = User.objects.filter(id__in=keys)
+           return Promise.resolve(sorted(users, key=lambda p: positions[p.id]))
```

Comment:
This won't work if you request a key that doesn't exist. The pattern we've been following is something like this:

```python
users = {user.id: user for user in User.objects.filter(id__in=keys)}
return Promise.resolve([users.get(id, None) for id in keys])
```